### PR TITLE
Show name, format and if uses descriptors in bitcoin-wallet tool

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

**Review by jonatack — COMMENTED** [review]
ACK a4b1588c6555b164fdd5e41164bf9358a5194660 except `test/functional/tool_wallet.py` needs to be updated

manual testing:
```
$ ./src/bitcoin-wallet -wallet=legacy info
Wallet info
===========
Name: legacy
Format: bdb
Descriptors: no
Encrypted: no
HD (hd seed available): yes
Keypool Size: 2000
Transactions: 28
Address Book: 54
```
```
$ ./src/bitcoin-wallet -wallet=new info
Wallet info
===========
Name: new
Format: sqlite
Descriptors: yes
Encrypted: no
HD (hd seed available): yes
Keypool Size: 6000
Transactions: 0
Address Book: 0
```

**Comment by jonatack** [comment]
Here's a minimal test update: https://github.com/jonatack/bitcoin/commits/pr-20198-tool_wallet-test

**Comment by practicalswift** [comment]
ACK a4b1588c6555b164fdd5e41164bf9358a5194660: patch looks correct :)

**Comment by promag** [comment]
Code review ACK, just needs to update test as @jonatack already pointed out.

**Comment by MarcoFalke** [comment]
Assigned 0.21. milestone for the same reason that #20125 got the milestone

**Comment by jonasschnelli** [comment]
pushed also @jonatack test (thanks!).

**Comment by MarcoFalke** [comment]
Needs to be squashed to not break git bisect with the test_runner ;)

**Comment by jonasschnelli** [comment]
Squashed.

**Comment by MarcoFalke** [comment]
review ACK fa7c585f9b306f31bcaf2ee492688e618f356108

File: src/wallet/wallettool.cpp [hunk 1]
```diff
@@ -95,6 +95,9 @@ static void WalletShowInfo(CWallet* wallet_instance)
     LOCK(wallet_instance->cs_wallet);
 
     tfm::format(std::cout, "Wallet info\n===========\n");
+    tfm::format(std::cout, "Name: %s\n", wallet_instance->GetName());
```
Comment (jonatack) [inline src/wallet/wallettool.cpp:98]: The linter is complaining about trailing whitespace in the test changes, but the whitespace is needed for the test to pass. Maybe print `Name:` or `Name: ""` when `wallet_instance->GetName().empty()`

Comment (MarcoFalke) [reply src/wallet/wallettool.cpp:98]: An alternative would be to write

```
    Name: \

    Format: bdb
```

in the test

Comment (jonasschnelli) [reply src/wallet/wallettool.cpp:98]: fixed.

Comment (jonatack) [reply src/wallet/wallettool.cpp:98]: still one left

`test/functional/tool_wallet.py:145:18: W291 trailing whitespace`

File: test/functional/tool_wallet.py [hunk 2]
```diff
@@ -98,12 +98,17 @@ def test_tool_wallet_info(self):
...
+
+            Format: bdb
+            Descriptors: no
             Encrypted: no
             HD (hd seed available): yes
             Keypool Size: 2
             Transactions: 0
             Address Book: 3
         ''')
+        print(out)
```
Comment (MarcoFalke) [inline test/functional/tool_wallet.py:111]: leftover debug print?

**Comment by jonatack** [comment]
ACK 67f100479270ca13d9380c8c9cfe45a0ddbaca7a 

The test now prints one of the test outputs.
```
$ test/functional/tool_wallet.py 
2020-10-21T11:24:29.497000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_9rm_zbea
2020-10-21T11:24:30.136000Z TestFramework (INFO): Testing that various invalid commands raise with specific error messages
2020-10-21T11:24:31.035000Z TestFramework (INFO): Calling wallet tool info, testing output
Wallet info
===========
Name: 
Format: bdb
Descriptors: no
Encrypted: no
HD (hd seed available): yes
Keypool Size: 2
Transactions: 0
Address Book: 3

2020-10-21T11:24:31.706000Z TestFramework (INFO): Generating transaction to mutate wallet
2020-10-21T11:24:32.014000Z TestFramework (INFO): Calling wallet tool info after generating a transaction, testing output
2020-10-21T11:24:32.165000Z TestFramework (INFO): Calling wallet tool create on an existing wallet, testing output
2020-10-21T11:24:34.577000Z TestFramework (INFO): Starting node with arg -wallet=foo
2020-10-21T11:24:35.587000Z TestFramework (INFO): Calling getwalletinfo on a different wallet ("foo"), testing output
2020-10-21T11:24:35.942000Z TestFramework (INFO): Check salvage
2020-10-21T11:24:37.247000Z TestFramework (INFO): Stopping nodes
2020-10-21T11:24:37.247000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_9rm_zbea on exit
2020-10-21T11:24:37.247000Z TestFramework (INFO): Tests successful
```

**Comment by jonasschnelli** [comment]
I removed the accidentally added test print.

**Comment by MarcoFalke** [comment]
ACK fa4074b395a47c54069bd9f598244701505ff11d

**Comment by jonatack** [comment]
re-ACK fa4074b395a47c54069bd9f598244701505ff11d

**Pull request merged**